src/reducers/console.js

```diff
@@ -17,85 +32,75 @@ function updateConsoleForHistoryIndex(state, index) {
 
   const expression = expressionHistory.get(index);
 
-  return state.
+  const nextState = state.
     set('historyEntryIndex', index).
     set('currentInputValue', expression);
-}
 
-function setNextConsoleEntry(state, historyIndex) {
-  const firstUp = historyIndex === 1;
+  if (index === 0) {
+    return nextState.delete('nextConsoleEntry');
+  }
 
-  if (firstUp && !state.history.isEmpty()) {
-    return state.set('nextConsoleEntry', state.currentInputValue);
+  if (isNil(state.nextConsoleEntry) && !state.history.isEmpty()) {
+    return nextState.set('nextConsoleEntry', state.currentInputValue);
   }
 
-  return state;
+  return nextState;
 }
 
-export default function console(stateIn, {type, payload, meta}) {
-  let state = stateIn;
-  if (state === undefined) {
-    state = initialState;
-  }
+export default handleActions({
+  [consoleValueProduced]: (
+    state,
+    {payload: {compiledProjectKey, key, value}},
+  ) => state.updateIn(
+    ['history', key],
+    input => input.set('value', value).
+      set('evaluatedByCompiledProjectKey', compiledProjectKey),
+  ),
+
+  [consoleErrorProduced]: (
+    state,
+    {payload: {compiledProjectKey, key, message, name}},
+  ) => state.updateIn(
+    ['history', key],
+    input => input.set('error', new ConsoleError({name, message})).
+      set('evaluatedByCompiledProjectKey', compiledProjectKey),
+  ),
 
-  switch (type) {
-    case 'CONSOLE_VALUE_PRODUCED':
-      return state.updateIn(
-        ['history', payload.key],
-        input => input.set(
-          'value',
-          payload.value,
-        ).set(
-          'evaluatedByCompiledProjectKey',
-          payload.compiledProjectKey,
-        ),
-      );
-    case 'CONSOLE_ERROR_PRODUCED':
-      return state.updateIn(
-        ['history', payload.key],
-        input => input.set(
-          'error',
-          new ConsoleError({name: payload.name, message: payload.message}),
-        ).set(
-          'evaluatedByCompiledProjectKey',
-          payload.compiledProjectKey,
-        ),
-      );
-    case 'EVALUATE_CONSOLE_ENTRY':
-      return payload.trim(' ') === '' ? state : state.setIn(
-        ['history', meta.key],
-        new ConsoleEntry({expression: payload}),
-      ).
+  [evaluateConsoleEntry]: (state, {payload: expression, meta: {key}}) =>
+    expression.trim(' ') === '' ?
```

Comment:
Is this a String `trim` method? If so it shouldn't require `' '` as an argument.